api/src/client/dynamics/elevator-safety/DynamicsElevatorSafetyRegistrationClient.test.ts

```diff
@@ -0,0 +1,150 @@
+import { DynamicsElevatorSafetyRegistrationClient } from "@client/dynamics/elevator-safety/DynamicsElevatorSafetyRegistrationClient";
+import { ElevatorSafetyRegistrationClient } from "@client/dynamics/elevator-safety/types";
+import { LogWriter, LogWriterType } from "@libs/logWriter";
+import axios from "axios";
+
+jest.mock("axios");
+jest.mock("winston");
+const mockAxios = axios as jest.Mocked<typeof axios>;
+
+describe("DynamicsElevatorSafetyRegistrationClient", () => {
+  let client: ElevatorSafetyRegistrationClient;
+  let logger: LogWriterType;
+
+  const ORG_URL = "www.test-org-url.com";
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    logger = LogWriter("NavigatorWebService", "ApiLogs", "us-test-1");
+
+    client = DynamicsElevatorSafetyRegistrationClient(logger, ORG_URL);
+  });
+
+  const mockAccessToken = "access-granted";
+  const propertyInterestId = "property-interest-id";
+
+  it("returns registration data on matching property interest", async () => {
+    const elevatorSafetyRegistrationMockResponse = {
+      value: [
+        {
+          ultra_buildingaddressline1: "address-1",
+          _ultra_propertyinterest_value: propertyInterestId,
+          ultra_requestdate: "2021-06-14T04:00:29Z",
+          ultra_elevatorregistrationrequestid: "request-id-1",
+          statuscode: "1",
+        },
+      ],
+    };
```

Comment:
[sand] In other places in the code we might make a generator for something like this. Given you use similar values for a mock response it might be helpful to create a generator that returns the response data you're expecting. See `api/test/factories.ts`.

Reply:
I'll add a refactor item to my icebox to tackle this on Friday